### ISSUE TITLE
Backups Review - need to review backups and recovery across infra

### ISSUE THREAD

**Issue by cconstab** [atsign-foundation/at_server#608]
Backups on infra need to be reviewed and checked to be recoverable.

Technology used likely could also be improved as GCP now has better options for GKE and Netvolumes

- [x] Secondaries NFS mount - existing snapshots + cron job to AWS
- [x] Root GKE Cluster - existing Velero job
- [x] Filebin2 GKE Cluster - unknown
- [x] filebin2 DB - unknown
- [x] Web Cluster one GKE cluster - existing Velero job
- [x] Web Cluster two GKE cluster - existin Velero job
- [x] Dev Secondaries NFS mount - cron job to AWS
- [x] Dev GKE cluster - existing Velero job
- [x] Other infra ?
- [x] Backups of MongoDB (registrar) - need to be checked too

This may not be a complete list... 

- [x]  Confirm and document solution in place or upgraded to XXX
- [x] Confirm recovery from backups successful

**Comment by gkc** [comment]
@kumarnarendra701 @athandle has there been any progress on this?

**Comment by athandle** [comment]
@gkc no progress on this ticket other tickets were added, will do in this sprint

**Comment by athandle** [comment]
@gkc @cpswan  this ticket will take more than 8SP, request you to approve and update the same

**Comment by cpswan** [comment]
@kumarnarendra701 after talking to @cconstab yesterday he's confirmed that he would like DNS etc. to be in scope for this.

**Comment by kumarnarendra701** [comment]
**K8 Backup Evidence -**
```
K8:Production Root -

narendrakumar@Narendras-MacBook-Pro-2 ~ % velero backup get | more
NAME                             STATUS            ERRORS   WARNINGS   CREATED                         EXPIRES    STORAGE LOCATION   SELECTOR
root-alone-prod-20220605070009   Completed         0        0          2022-06-05 12:30:09 +0530 IST   29d        default            <none>
root-alone-prod-20220605064509   Completed         0        0          2022-06-05 12:15:09 +0530 IST   29d  

K8:Ludo -

narendrakumar@Narendras-MacBook-Pro-2 ~ % velero backup get
NAME                            STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
ludo-k8-backup-20220605071516   Completed   0        0          2022-06-05 12:45:16 +0530 IST   29d       default            <none>
ludo-k8-backup-20220605071243   Completed   0        0          2022-06-05 12:42:43 +0530 IST   29d       default            <none>

K8:cluster-1 -

narendrakumar@Narendras-MacBook-Pro-2 ~ % velero backup get
NAME                      STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
web-prod-20220605070036   Completed   0        12         2022-06-05 12:30:36 +0530 IST   29d       default            <none>
web-prod-20220604070036   Completed   0        12         2022-06-04 12:30:36 +0530 IST   28d       default            <none>
web-prod-20220603070036   Completed   0        12         2022-06-03 12:30:36 +0530 IST   27d       default            <none>

K8:cluster-2 -

narendrakumar@Narendras-MacBook-Pro-2 ~ % velero backup getNAME                     STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
web-dev-20220605070035   Completed   0        0          2022-06-05 12:30:35 +0530 IST   29d       default            <none>
web-dev-20220604070035   Completed   0        0          2022-06-04 12:30:35 +0530 IST   28d       default            <none>




Filebin:K8

narendrakumar@Narendras-MacBook-Pro-2 atsign % velero backup getNAME                               STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
filebin-k8-backup-20220612121528   Completed   0        0          2022-06-12 17:45:28 +0530 IST   29d       default            <none>
```
**Filebin:SQL**

![image](https://user-images.githubusercontent.com/29299536/173285746-8e029971-d32f-4c4f-9380-0549d5efce2d.png)

**GCP added a new feature GKE Backup, so enabled for the below cluster.**
1. Development
2. Production root
3. Web Infrastructure(Cluster 1 and Cluster2)

Backup Evidence -

![image](https://user-images.githubusercontent.com/29299536/173285574-46a190a9-65a9-44e7-872c-119742c8acf7.png)

**NFS Backup Evidence -**
![image](https://user-images.githubusercontent.com/29299536/173286023-f77dda63-cc64-4f43-8c0d-b608f93c9721.png)

**K8 Restore Backup Evidence**
```
narendrakumar@Narendras-MacBook-Pro-2 atsign % velero install --provider aws --plugins velero/velero-plugin-for-aws:v1.4.1 --bucket atsign-ludo-backups --backup-location-config region=us-west-1 --snapshot-location-config region=us-west-1 --secret-file ./credentials-velero --use-restic

CustomResourceDefinition/backups.velero.io: attempting to create resource
CustomResourceDefinition/backups.velero.io: attempting to create resource client
CustomResourceDefinition/backups.velero.io: created
CustomResourceDefinition/backupstoragelocations.velero.io: attempting to create resource
CustomResourceDefinition/backupstoragelocations.velero.io: attempting to create resource client
CustomResourceDefinition/backupstoragelocations.velero.io: created
CustomResourceDefinition/deletebackuprequests.velero.io: attempting to create resource
CustomResourceDefinition/deletebackuprequests.velero.io: attempting to create resource client
CustomResourceDefinition/deletebackuprequests.velero.io: created
CustomResourceDefinition/downloadrequests.velero.io: attempting to cre



narendrakumar@Narendras-MacBook-Pro-2 atsign % velero restore create --from-backup ludo-k8-backup-20220612114518 --include-namespaces registrar-api-buzz
Restore request "ludo-k8-backup-20220612114518-20220612174419" submitted successfully.
Run `velero restore describe ludo-k8-backup-20220612114518-20220612174419` or `velero restore logs ludo-k8-backup-20220612114518-20220612174419` for more details.
narendrakumar@Narendras-MacBook-Pro-2 atsign % velero restore get
NAME                                           BACKUP                          STATUS      STARTED                         COMPLETED                       ERRORS   WARNINGS   CREATED                         SELECTOR
ludo-k8-backup-20220612114518-20220612174419   ludo-k8-backup-20220612114518   Completed   2022-06-12 17:44:20 +0530 IST   2022-06-12 17:44:31 +0530 IST   0        1          2022-06-12 17:44:20 +0530 IST   <none>
narendrakumar@Narendras-MacBook-Pro-2 atsign % kubectl get ns
NAME                 STATUS   AGE
default              Active   43h
kube-node-lease      Active   43h
kube-public          Active   43h
kube-system          Active   43h
registrar-api-buzz   Active   17s
velero               Active   2m28s
narendrakumar@Narendras-MacBook-Pro-2 atsign % kubectl get po -n registrar-api-buzz
NAME                                  READY   STATUS              RESTARTS   AGE
infrastructure-api-6b694b59b6-thlg7   0/1     ContainerCreating   0          23s
registrar-node-5d48fc9cf9-4zxlb       0/1     ContainerCreating   0          21s
root-manager-75579cdf8d-kgbwq         0/1     ContainerCreating   0          20s
narendrakumar@Narendras-MacBook-Pro-2 atsign % kubectl get po -n registrar-api-buzz
NAME                                  READY   STATUS              RESTARTS   AGE
infrastructure-api-6b694b59b6-thlg7   1/1     Running             0          27s
registrar-node-5d48fc9cf9-4zxlb       0/1     ContainerCreating   0          25s
root-manager-75579cdf8d-kgbwq         0/1     ContainerCreating   0          24s
narendrakumar@Narendras-MacBook-Pro-2 atsign % kubectl get po -n registrar-api-buzz
NAME                                  READY   STATUS    RESTARTS   AGE
infrastructure-api-6b694b59b6-thlg7   1/1     Running   0          36s
registrar-node-5d48fc9cf9-4zxlb       1/1     Running   0          34s
root-manager-75579cdf8d-kgbwq         1/1     Running   0          33s
narendrakumar@Narendras-MacBook-Pro-2 atsign % 
```

**Comment by kumarnarendra701** [comment]
For DO DNS Backup - Added shell script in prod swarm0001-01 to create a backup of all zone records and push it to S3.
```
# DO DNS Backup
45 11 * * * /opt/automated-dns-backup/do-dns-backup.sh >/dev/null 2>&1

Backup Evidence -
root@swarm0001-01:~# s3cmd ls s3://atsign-dns-backup
2022-06-13 08:04    643113   s3://atsign-dns-backup/dns_backup_2022-06-13_08-04-24.tar.gz
2022-06-13 08:13       890   s3://atsign-dns-backup/dns_backup_2022-06-13_08-13-11.tar.gz
2022-06-13 08:16    643208   s3://atsign-dns-backup/dns_backup_2022-06-13_08-15-39.tar.gz
```


**For Upgrades -** 
NFS - Not required
K8 - There are no major updates required.

**Comment by kumarnarendra701** [comment]
Notes- If we require Namecheap to schedule DNS backup then please share and grant me access. So that I'll create a new ticket and work on this.

@cpswan @athandle - Can we close this?

**Comment by cpswan** [comment]
@kumarnarendra701 I don't think we're quite ready to close this, but also I think your work is done (at least for now).

I'd still like to get @cconstab view on some of the other resources and our (lack of) backup approach for them.

**Comment by kumarnarendra701** [comment]
@cpswan - Please let me know if I need to do anything or if we can close this.

**Comment by cpswan** [comment]
We can't close yet. I've still got to have that chat with @cconstab

**Comment by cpswan** [comment]
@cconstab you were going to add some resources here for DNS backup.

Moving to PR41 as we're still not quite done with this.

**Comment by cconstab** [comment]
Some examples
https://www.cloudfloordns.com/
https://dnsmadeeasy.com/services/dnsfailover

We 'just' need to secondary the DNS zones we use for secondaries I think

**Comment by cconstab** [comment]
Spotted this failure in our backups as I was reviewing this ticket..

#799

**Comment by cconstab** [comment]
MongoDB backups also need to be checked please @kumarnarendra701

**Comment by athandle** [comment]
@cconstab I think you already have that backup setup in place on cloud

**Comment by cpswan** [comment]
@athandle I think @cconstab wants to check that we can restore something from those backups (as @kumarnarendra701 has tested with all the others on the checklist).

**Comment by cpswan** [comment]
@cconstab as I was looking at Secondary DNS options I noticed stuff that reminded me of NS1, so I took a look at whether they can do secondary DNS, and they do. So in the spirit of not adding /another/ vendor should we just use NS1 for secondary DNS?

**Comment by cconstab** [comment]
Great idea..

**Comment by cpswan** [comment]
@cconstab I went to configure NS1 as a secondary DNS, and pretty much fell at the first hurdle as Digital Ocean don't appear to support AXFR for zone transfers.

**Comment by cconstab** [comment]
Doh

**Comment by cpswan** [comment]
@athandle @kumarnarendra701 did the MongoDB backups get checked yet?

We've decided not to do anything more with DNS (for now) so once MongoDB is confirmed we can get this closed.

**Comment by cpswan** [comment]
@athandle @kumarnarendra701 we're still waiting on confirmation about MongoDB backups before this can be closed.

**Comment by kumarnarendra701** [comment]
@cpswan - We have daily and hourly backup configured for atlas databases and the option to recover specific snapshots.

![image](https://user-images.githubusercontent.com/29299536/183404099-f49d8102-7cee-4ecd-992b-4257a50f69b1.png)

**Comment by cpswan** [comment]
Thanks @kumarnarendra701 

Have we tested restoring one of those backups?

**Comment by gkc** [comment]
Moving to PR43 as still ongoing

**Comment by athandle** [comment]
mongo backup restore tested
on click of restore button, backup is restoring
![image](https://user-images.githubusercontent.com/61007675/184850413-d873606a-b6a6-49bf-bd4a-cfec6ade36d4.png)
